packages/backend/src/modules/verifiers/VerifiersController.ts

```diff
@@ -0,0 +1,111 @@
+import { Logger } from '@l2beat/backend-tools'
+import {
+  Layer2,
+  ZkCatalogProject,
+  layer2s,
+  zkCatalogProjects,
+} from '@l2beat/config'
+import {
+  assert,
+  EthereumAddress,
+  VerifiersApiResponse,
+  cacheAsyncFunction,
+} from '@l2beat/shared-pure'
+import { Project } from '../../model/Project'
+import { BlockscoutClient } from '../../peripherals/blockscout/BlockscoutClient'
+import { TaskQueue } from '../../tools/queue/TaskQueue'
+
+export interface VerifiersControllerDeps {
+  blockscoutClient: BlockscoutClient
+  projects: Project[]
+  logger?: Logger
+}
+
+export class VerifiersController {
+  private readonly taskQueue: TaskQueue<void>
+  private readonly logger: Logger
+  getCachedVerifierStatuses: () => Promise<VerifiersApiResponse>
+
+  constructor(private readonly $: VerifiersControllerDeps) {
+    this.logger = $.logger ? $.logger.for(this) : Logger.SILENT
+
+    const cached = cacheAsyncFunction(() => this.getVerifierStatuses())
+    this.getCachedVerifierStatuses = cached.call
+    this.taskQueue = new TaskQueue(
+      cached.refetch,
+      this.logger.for('taskQueue'),
+      { metricsId: VerifiersController.name },
+    )
+  }
+
+  start() {
+    this.taskQueue.addToFront()
+    this.logger.info('Caching: initial caching scheduled')
+
+    const hour = 60 * 60 * 1000
+    setInterval(() => {
+      this.taskQueue.addIfEmpty()
+      this.logger.info('Caching: refetch scheduled')
+    }, hour)
+  }
+
+  async getVerifierStatuses(): Promise<VerifiersApiResponse> {
+    const addresses = this.getVerifierAddresses()
+    assert(addresses.length > 0, 'No verifier addresses found')
+
+    return await Promise.all(
+      addresses.map(async (address) => {
+        try {
+          const txs =
+            await this.$.blockscoutClient.getInternalTransactions(address)
+          txs.sort((a, b) => b.timestamp.toNumber() - a.timestamp.toNumber())
```

Comment:
blockscout returns random order?

Reply:
it is sorted but we have no control over it (no sorting param, etc.) so it's better to stay on the safe side here